lib/fastlane/actions/clear_derived_data.rb

```diff
@@ -16,11 +22,16 @@ def self.description
       end
 
       def self.details
-        "Deletes the Derived Data from '~/Library/Developer/Xcode/DerivedData'"
+        "Deletes the Derived Data from '~/Library/Developer/Xcode/DerivedData' or a supplied path"
       end
 
       def self.available_options
-        []
+        [
+          FastlaneCore::ConfigItem.new(key: :derived_data_path,
+                                       env_name: "DERIVED_DATA_PATH",
+                                       description: "Custom path for derivedData",
```

Comment:
This should have `default_value: "~/Library/Developer/Xcode/DerivedData"` and then expand_path at a later point